lib/client/client.go

```diff
@@ -108,20 +111,45 @@ func (c *Client) EnsureInitialized() error {
 		return err
 	}
 
-	// Ensure a cluster GUID is set.  This is required for all datastore
-	// types.
-	kv := &model.KVPair{
-		Key:   model.GlobalConfigKey{Name: "ClusterGUID"},
-		Value: hex.EncodeToString(uuid.NewV4().Bytes()),
+	// Ensure a number of global Felix config values are configured:
+	// -  GUID (ClusterGUID)
+	// -  IPIP (IpInIpEnabled)
+	// -  Felix reporting interval (ReportingIntervalSecs)
+	// -  Log severity for screen (LogSeverityScreen)
+	// -  Log severity for file (LogSeverityFile)
+	// -  Log file path (LogFilePath)
+	guid := hex.EncodeToString(uuid.NewV4().Bytes())
+	if err := c.ensureGlobalFelixConfig("ClusterGUID", guid); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("ReportingIntervalSecs", "0"); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("IpInIpEnabled", GlobalDefaultIPIP); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("LogSeverityScreen", GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("LogSeverityFile", "none"); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("LogFilePath", GlobalDefaultLogLevel); err != nil {
+		return err
 	}
-	if _, err := c.backend.Create(kv); err == nil {
-		log.WithField("ClusterGUID", kv.Value).Info("Assigned ClusterGUID")
-	} else {
-		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
-			log.WithError(err).Warnf("Failed to set ClusterGUID")
-			return err
-		}
-		log.Info("Using previously configured ClusterGUID")
+
+	// Ensure a number of global BGP config values are configured:
+	// -  Node to node mesh (node_mesh)
+	// -  Log severity (loglevel)
+	// -  Default node AS number
+	nm, _ := json.Marshal(nodeToNodeMesh{Enabled: GlobalDefaultNodeToNodeMesh})
```

Comment:
Presumably confd falls over if these don't exist?  That's probably fine for now then.